src/spirc.rs

```diff
@@ -128,6 +198,11 @@ impl SpircInternal {
             self.last_command_ident = frame.get_ident().to_owned();
             self.last_command_msgid = frame.get_seq_nr();
         }
+
+        if frame.has_ident() && !frame.has_goodbye() && frame.has_device_state() {
+            self.devices.insert(frame.take_ident(), frame.take_device_state().take_name());
```

Comment:
I'd rather use `get_` rather than `take_`.
Take is a small optimization, but it may cause confusion in the future where the ident and device state are missing from the object.